nanofi/src/api/nanofi.cpp

```diff
@@ -223,13 +226,15 @@ void free_flowfile(flow_file_record *ff) {
     return;
   }
   auto content_repo_ptr = static_cast<std::shared_ptr<minifi::core::ContentRepository>*>(ff->crp);
-  if (content_repo_ptr->get()) {
+  if (content_repo_ptr->get() && (ff->keepContent == 0)) {
```

Comment:
@phrocker : I don't link this, but I have no better idea yet. We provide a flow_file_record to access content and attributes in the processor logic callback. This means that the content cannot be deleted in case the flowfile is transfered via a relationship.

Reply:
Yeah that seems like it could cause issues when freeing the flow file, but if they were to do all of this, this means that they've either threaded the reads of content or are calling free in the middle of accessing that content? I think it's reasonable for each read to "keep the content" because it's not the owner. A better way might be to use the reference counting in ResourceClaim but we don't presently have that since we make new claim objects. I think what you've done is not unreasonable.